File: C/section07/S07E07.c
#include <stdio.h>

int main() { 
    int mouseId;
    int mouseQuantity = 0, mouseState = 0, needsSphere = 0, needsCleaning = 0, needsCableOrConectorChange = 0, brokenOrUseless = 0;
    float needsSpherePercentage, needsCleaningPercentage, needsCableOrConectorChangePercentage, brokenOrUselessPercentage;

    printf("Please enter the mouse Id: \n");
    scanf("%d", &mouseId);

    while(mouseId != 0) {
        
        printf("Please enter the mouse state: \n1 - needs sphere.\n2 - needs cleaning.\n3 - needs cable or conector change.\n4 - It's broken or useless.\n");
        scanf("%d", &mouseState);

        if (mouseState <= 4 && mouseState >= 1) {
            if (mouseState == 1) {
            needsSphere += +1;
            } else if (mouseState == 2) {
            needsCleaning += +1;
            } else if (mouseState == 3) {
            needsCableOrConectorChange += +1;
            } else {
            brokenOrUseless += +1;
            }

            mouseQuantity++;
            printf("\nMouse state of ID %d saved succesfully.\n", mouseId);
        } else {
        printf("\nMouse state not found. Please try again.\n");
        }

    printf("Please enter the mouse Id: \n");
    scanf("%d", &mouseId);
        
    }
    
    if (mouseQuantity != 0) {
    needsSpherePercentage = ((float) needsSphere / (float) mouseQuantity * 100.0);
    needsCleaningPercentage = ((float) needsCleaning / (float) mouseQuantity * 100.0);
    needsCableOrConectorChangePercentage = ((float) needsCableOrConectorChange / (float) mouseQuantity * 100.0);
    brokenOrUselessPercentage = ((float) brokenOrUseless / (float) mouseQuantity * 100.0);
    }

    printf("\nMouse quantity: %d\n\n", mouseQuantity);
    printf("Mouse state  \t\t\t\t\tQuantity \t\tPercentage\n");
    printf("1 - Needs sphere  \t\t\t\t%d \t\t\t%.2f%%\n", needsSphere, needsSpherePercentage);
    printf("2 - Needs cleaning  \t\t\t\t%d \t\t\t%.2f%%\n", needsCleaning, needsCleaningPercentage);
    printf("3 - Needs cable or conector change \t\t%d \t\t\t%.2f%%\n", needsCableOrConectorChange, needsCableOrConectorChangePercentage);
    printf("4 - Broken or useless  \t\t\t\t%d \t\t\t%.2f%%\n", brokenOrUseless, brokenOrUselessPercentage);
    
    return 0;    
}

File: Python/section07/S07e07.py
mouseQuantity = 0
needsSphere = 0
needsCleaning = 0
needsConectorOrCableChange = 0
brokenOrUseless = 0
needsSpherePercentage = 0
needsCleaningPercentage = 0
needsConectorOrCableChangePercentage = 0
brokenOruselessPercentage = 0

mouseCode = int(input("\nPlease enter a mouse code, enter 0 to show results: "))

if mouseCode != 0:
    while mouseCode != 0:
        
        optionCode = int(input("\nPlease choose an option:\n1 - needs sphere.\n2 - needs cleaning\n3 - needs conector or cable change\n4 - broken or useless\n"))
        
        if optionCode > 1 or optionCode < 4:
            while optionCode < 1 or optionCode > 4:
                optionCode = int(input("\nNo such code, please choose between 1 and 4: \n1 - needs sphere.\n2 - needs cleaning\n3 - needs conector or cable change\n4 - broken or useless\n"))
     
        if optionCode == 1:
            needsSphere += 1
        elif optionCode == 2:
            needsCleaning += 1
        elif optionCode == 3:
            needsConectorOrCableChange += 1
        else:
            brokenOrUseless += 1
        
        mouseQuantity += 1
        print("\nMouse status saved.\n")
        mouseCode = int(input("\nPlease enter a mouse code, enter 0 to show results: "))

    if mouseQuantity != 0:
        needsSpherePercentage = (needsSphere / mouseQuantity) * 100
        needsCleaningPercentage = (needsCleaning / mouseQuantity) * 100
        needsConectorOrCableChangePercentage = (needsConectorOrCableChange / mouseQuantity) * 100
        brokenOruselessPercentage = (brokenOrUseless / mouseQuantity) * 100

print("\nMouse quantity: {0}\n".format(mouseQuantity))
print("Status                             Quantity    Percentage")
print("1 - needs sphere                   {0}           {1:.2f}%".format(needsSphere, needsSpherePercentage))
print("2 - needs cleaning                 {0}           {1:.2f}%".format(needsCleaning, needsCleaningPercentage))
print("3 - needs conect or cable change   {0}           {1:.2f}%".format(needsConectorOrCableChange, needsConectorOrCableChangePercentage))
print("4 - broken or useless              {0}           {1:.2f}%".format(brokenOrUseless, brokenOruselessPercentage))